src/edits/performEdits.ts

```diff
@@ -79,7 +91,9 @@ export function performEdits(origin, ast, edits, onSuccess=()=>{}, onError=()=>{
         group.completed = true;
       }
     } else {
-      changeArray.push(edit.toChangeObject(ast));
+      if (edit.toChangeObject) {
+        changeArray.push(edit.toChangeObject(ast));
+      }
```

Comment:
This is the one potential behavior change. Only some `Edit` subclasses implement a `toChangeObject` method. Hopefully that's ok. To make the types work, I had to add a check for the existence of `toChangeObject` before calling it.

Reply:
This is the one area where I'm scratching my head a little. It's been a long time since I dug into this code, and I want to convince myself that there isn't a problem with the fact that some edits don't implement this method. This doesn't seem to break any tests, so I'll approve it and keep reading until I'm convinced. :)